.design/requirements-587.md

```diff
@@ -0,0 +1,101 @@
+Endpoint Implementor - a portion of the code that can be used to register an endpoint with the API
+
+## Preliminary work
+
+An implementation of the proposed specification can be found here:
+https://github.com/stbischof/workground/tree/main/ws
+
+## Problem Description
+
+While in a traditional web application the configuration is rather static and known in advance in OSGi items can come and go anytime and code should be aware of this dynamism.
+Also registration of listeners must happen before an endpoint is published and only 
+weak typing makes it hard to hanlde this righ manually.
+
+As a result, the publishing of Web Services in the current state of the specification goes against the loose coupling concept usually provided by OSGi 
+as it requires knowledge about the stakeholders involved, beside that it is a non trivial 
+task to integrate with the OSGi Whiteboard Specification for Jakarta™ Servlet as additional 
+actions and care must be taken here.
+
+It is therefore the aim of this specification to handle this cases in a transparent manner so the user can focus 
+on the actuall work, use common dependency injection techniques like Declarative Services 
+to inject dependecies.
+
+## Use Cases
+
+The code below shows an example of a logging handler that wants to intercep incomming 
+and outgoing messages, this handler is targeted to any registered Endpoint Implementor:
+
+```java
+@Component(service = Handler.class)
+public class SOAPLoggingHandler implements SOAPHandler<SOAPMessageContext> {
+
+	public Set<QName> getHeaders() {
+		return null;
+	}
+
+	public boolean handleMessage(SOAPMessageContext smc) {
+		try {
+			smc.getMessage().writeTo(System.out);
+		} catch (SOAPException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return true;
+	}
+
+	public boolean handleFault(SOAPMessageContext smc) {
+		try {
+			smc.getMessage().writeTo(System.err);
+		} catch (SOAPException e) {
+			e.printStackTrace();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return true;
+	}
+
+	public void close(MessageContext messageContext) {
+	}
+}
+```
+
+The code below shows an simple echo webservice Endpoint Implementor, it uses a relative 
+URI and is therefore targeted to the http whiteboard service, the `@WebService` / `@WebMethod` is 
+a jakarta annotation:
+
+```java
+@WebService
+@SOAPWhiteboardEndpoint(contextpath = "/echo")
+@Component(immediate = true, service = WSEcho.class)
+public class WSEcho {
+
+	@WebMethod(operationName = "echo", action = "echo")
+	public String echo(@WebParam(name = "textIn") String text) {
+		return text;
+	}
+
+}
+```
+
+## Requirements
+
+- WS-0010: The solution MUST make it possible to register an Endpoint Implementor to a specific endpoint address.
+- WS-0020: The solution MUST make it possible to register an Endpoint Implementor to the Whiteboard Specification for Jakarta™ Servlet.
```

Comment:
please remove this as discussed in spec call

Reply:
Done.